src/CP/variableselection/failureBased.jl

```diff
@@ -0,0 +1,30 @@
+struct FailureBasedVariableSelection{TakeObjective} <: AbstractVariableSelection{TakeObjective} end
+
+FailureBasedVariableSelection(;take_objective=true) = FailureBasedVariableSelection{take_objective}()
+
+function (::FailureBasedVariableSelection{false})(cpmodel::CPModel)
+    selectedVar = nothing
+    minSize = Inf
+    for (k, x) in branchable_variables(cpmodel)
+        if x !== cpmodel.objective && !isbound(x) && length(x.domain) / cpmodel.statistics.infeasibleStatusPerVariable[id(x)] <= minSize
+            selectedVar = x
+            minSize = length(x.domain) / cpmodel.statistics.infeasibleStatusPerVariable[id(x)]
```

Comment:
![image](https://user-images.githubusercontent.com/59280588/124324120-731bcb00-db50-11eb-874e-608a2d81d227.png)

Reply:
This is the core function to review.